src/backend/distributed/commands/vacuum.c

```diff
@@ -484,6 +502,30 @@ VacuumStmtParams(VacuumStmt *vacstmt)
 			params.truncate = defGetBoolean(opt) ? VACOPT_TERNARY_ENABLED :
 							  VACOPT_TERNARY_DISABLED;
 		}
+		#if PG_VERSION_NUM >= PG_VERSION_13
+		else if (strcmp(opt->defname, "parallel") == 0) {
+			
+			if (opt->arg == NULL)
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_SYNTAX_ERROR),
+						 errmsg("parallel option requires a value between 0 and %d",
+								MAX_PARALLEL_WORKER_LIMIT)));
+			}
+			else
+			{
+				int			nworkers;
+				nworkers = defGetInt32(opt);
+				if (nworkers < 0 || nworkers > MAX_PARALLEL_WORKER_LIMIT)
+					ereport(ERROR,
+							(errcode(ERRCODE_SYNTAX_ERROR),
+							 errmsg("parallel vacuum degree must be between 0 and %d",
+									MAX_PARALLEL_WORKER_LIMIT)));
+
+				params.nworkers = nworkers;
+			}
+		}
+		#endif
```

Comment:
As the comment of this function describes:

```C
/*
 * This is mostly ExecVacuum from Postgres's commands/vacuum.c
 */
```

I actually was wondering why we are copying this code from postgres instead of relying on postgres to provide the errors.

My guess is that we don't call postgres' Vacuum, as we assume distributed tables don't have anything to vacuum on the coordinator. However this could be very well a wrong assumption.

With your insights, would you be able to describe in the comment on this function the _why_ of this function being almost an exact copy?

Reply:
b0baa27